xai_components/utils.py

```diff
@@ -41,7 +42,10 @@ def set_defaults(component, tvb_datatype):
     for attr in tvb_datatype.declarative_attrs:
         attr_value = getattr(tvb_datatype, attr)
         attr_default = attr_value.default
-        if isclass(attr_default) and issubclass(attr_value.field_type, HasTraits):
+        print(type(attr_default))
+        if issubclass(tvb_datatype, Equation) and callable(attr_default) and attr_default.__name__ == '<lambda>':
+            attr_default = attr_default()
+        elif isclass(attr_default) and issubclass(attr_value.field_type, HasTraits):
             attr_default = attr_default()
```

Comment:
both of these branches do the same thing. Was this intended ?

Reply:
the first one is specifically targeted for the `parameters` attribute of `Equation`. the second one is more general and refers to all the attributes that have classes as default values (ex. the noise attribute from an IntegratorStochastic has the `Additive` class provided as default value). but yes, both branches do the same thing, so I guess I could merge them into one longer `if` condition.